src/geobr.jl

```diff
@@ -0,0 +1,590 @@
+# ------------------------------------------------------------------
+# Licensed under the MIT License. See LICENSE in the project root.
+# ------------------------------------------------------------------
+"""
+Provides functions to download data from the GeoBR database:
+
+`GeoBR.extractmetadata`
+`GeoBR.download`
+`GeoBR.get`
+`GeoBR.state`
+`GeoBR.municipality`
+`GeoBR.region`
+`GeoBR.country`
+`GeoBR.amazon`
+`GeoBR.biomes`
+`GeoBR.disasterriskarea`
+`GeoBR.healthfacilities`
+`GeoBR.indigenousland`
+`GeoBR.metroarea`
+`GeoBR.neighborhood`
+`GeoBR.urbanarea`
+`GeoBR.weightingarea`
+`GeoBR.mesoregion`
+`GeoBR.microregion`
+`GeoBR.intermediateregion`
+`GeoBR.immediateregion`
+`GeoBR.municipalseat`
+`GeoBR.censustract`
+`GeoBR.statisticalgrid`
+`GeoBR.conservationunits`
+`GeoBR.semiarid`
+`GeoBR.schools`
+`GeoBR.comparableareas`
+`GeoBR.urbanconcentrations`
+`GeoBR.poparrangements`
+`GeoBR.healthregion`
+`GeoBR.filepath`
+"""
+module GeoBR
+
+using GeoIO
+using DataDeps
+using CSV
+using Tables
+using TableTransforms
+
+const APIVERSIONS = (v"1.7.0",)
+
+function metadata()
+    url = "http://www.ipea.gov.br/geobr/metadata/metadata_1.7.0_gpkg.csv"
+    ID = "GeoBR_metadata"
+    path = ""
+    try
+        path = @datadep_str filepath(ID, url)
+    catch
+        register(DataDep(ID,
+            """
+            Metadata for GeoBR package.
+            Source: $url
+            """,
+            url,
+            Any
+        ))
+        path = @datadep_str filepath(ID, url)
+    end
+    return path
+end
+
+function extractmetadata(geo, year, code, abbrev; returnall=false)
+    path = metadata()
+    table = CSV.File(path)
+
+    srows = table |> Filter(row ->
+        row.geo == geo &&
+            row.year == year &&
+            (isnothing(code) || parse(Int, row.code) == code) &&
+            (isnothing(abbrev) || row.code_abbrev == abbrev)
+    )
+
+    if isempty(srows)
+        throw(ErrorException("No matching rows found for the given parameters"))
+    end
+
+    return returnall ? Tables.rows(srows) : Tables.rows(srows) |> first
+end
+
+function download(geo, year, code, abbrev)
+    srow = extractmetadata(geo, year, code, abbrev)
```

Comment:
the function continues using extractmetadata, it should use metadata instead as suggested before